packages/search/src/users/users.ts

```diff
@@ -0,0 +1,146 @@
+/* Copyright (c) 2021 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+
+import { getProp } from "@esri/hub-common";
+import { UserSession } from "@esri/arcgis-rest-auth";
+
+import { GraphQLClient } from "graphql-request";
+import { GraphQLError } from "./errors";
+import {
+  createSessionMutation,
+  userSelfQuery,
+  userSearchQuery
+} from "./queries";
+
+import { IUser } from "@esri/arcgis-rest-types";
+import {
+  base64,
+  IDateRange,
+  SortDirection,
+  ICursorSearchResults
+} from "../types";
+
+export enum UserSortableField {
+  USERNAME = "USERNAME",
+  LAST_HUB_SESSION = "LAST_HUB_SESSION"
+}
+
+interface ISortingOption {
+  field?: UserSortableField;
+  sortDirection?: SortDirection;
+}
+
+interface IEdge {
+  node: IHubUser;
+  cursor?: base64;
+}
+
+export interface IHubUser extends IUser {
+  followedInitiatives?: any[];
+  groups?: any[];
+  teams?: any[];
+  registeredEvents?: any[];
+  lastHubSession?: number;
+  visitsLast30Days?: number;
+  visitsLast60Days?: number;
+}
+
+export interface ISearchUsersFilter {
+  lastHubSession?: IDateRange;
+  group?: string;
+  team?: string;
+  followedInitiative?: string;
+  registeredEvent?: string;
+}
+
+export interface IUserPagingOptions {
+  first: number;
+  after?: string;
+}
+
+export interface IUserSearchOptions {
+  pagingOptions?: IUserPagingOptions;
+  sortingOptions?: ISortingOption[];
+}
+
+export class UserService {
+  constructor(private portalUrl: string, private userIndex: GraphQLClient) {}
+
+  static create(
+    portalUrl: string,
+    userIndexApi: string,
+    authentication: UserSession
+  ): UserService {
+    const api: GraphQLClient = new GraphQLClient(userIndexApi, {
```

Comment:
I am creating the client in the static function and passing it directly to the constructor so that we can stub this API object easily in tests